src/state/opamEnv.ml

```diff
@@ -344,12 +370,13 @@ let eval_string gt ?(set_opamswitch=false) switch =
         OpamFilename.Dir.to_string OpamStateConfig.(default.root_dir)
       ) in
     if opamroot_cur <> opamroot_env then
-      Printf.sprintf " --root=%s" opamroot_cur
+      Some opamroot_cur
     else
-      "" in
+      None
+  in
   let switch =
     match switch with
```

Comment:
This can be refactored slightly more clearly as:
```ocaml
let f sw =
  let sw_cur = ... (* etc. code from the `Some sw` case *)
in
OpamStd.Option.replace f switch
```

Reply:
Thanks, refactored